Sources/SwiftDocC/Checker/Checker.swift

```diff
@@ -270,7 +270,7 @@ public struct CompositeChecker: Checker {
     public var checkers: [AnyChecker]
 
     /// Creates a checker that performs the combined work of the given checkers.
-    public init<Checkers: Sequence>(_ checkers: Checkers) where Checkers.Element: Checker {
+    public init(_ checkers: some Sequence<Checker>) {
```

Comment:
Found a few more places where we could simplify the generic syntax using `some`:

SemanticAnalysis.swift
```
func analyze<Children: Sequence>(
  _ directive: BlockDirective,
  children: Children, source: URL?,
  for bundle: DocumentationBundle,
  in context: DocumentationContext, problems: inout [Problem]
) -> Result where Children.Element == Markup
```

to:
```
func analyze(
   _ directive: BlockDirective,
  children: some Sequence<Markup>,
  source: URL?,
  for bundle: DocumentationBundle,
  in context: DocumentationContext, problems: inout [Problem]
) -> Result
```




DocumentationConverter.swift
```
mutating func convert<OutputConsumer: ConvertOutputConsumer>(
  outputConsumer: OutputConsumer
) throws -> (analysisProblems: [Problem], conversionProblems: [Problem])
```

to:
```
mutating func convert(
  outputConsumer: some ConvertOutputConsumer
) throws -> (analysisProblems: [Problem], conversionProblems: [Problem])
```

DirectiveArgumentWrapper.swift
     
```
func setProperty<T: AutomaticDirectiveConvertible>(
  on containingDirective: T,
  named propertyName: String,
  to any: Any
)
```

to:
```
func setProperty(
  on containingDirective: some AutomaticDirectiveConvertible,
  named propertyName: String,
  to any: Any
)
```


There are many other places in the code where we have generic constraints that appear to be places where we could add `some` - but actually don't compile. The most common example of these are the places where we use the visitor pattern:


Symbol.swift
```
public override func accept<V: SemanticVisitor>(_ visitor: inout V) -> V.Result {
   return visitor.visitSymbol(self)
}
```

Here it appears we could write 
```
public override func accept(_ visitor: inout some SemanticVisitor) -> SemanticVisitor.Result {
   return visitor.visitSymbol(self)
}
```

.... but Swift returns an error: `Cannot access associated type 'Result' from 'SemanticVisitor'; use a concrete type or generic parameter base instead`

Any idea why? Why can't Swift access the `Result` associated type? Is there a better way to write the function declaration in this example?

Reply:
I didn't realize that I could update the declaration in the protocols themselves. I got some error message about that earlier and thought that it wasn't supported but it looks like it is.

Reply:
> .... but Swift returns an error: Cannot access associated type 'Result' from 'SemanticVisitor'; use a concrete type or generic parameter base instead

`V.Result` is referring to the specific `Result` associated type of `V`. However `SemanticVisitor.Result` isn't connected to the `some SemanticVisitor` parameter, so it doesn't define a return type that's associated with the parameter type.

Reply:
> V.Result is referring to the specific Result associated type of V. However SemanticVisitor.Result isn't connected to the some SemanticVisitor parameter, so it doesn't define a return type that's associated with the parameter type.

Ok. But don't all types that conform to `SemanticVisitor` have the same associated type `Result`? It seems to me that Swift should be able to make this compile and run.